logger/tests/integration_tests.rs

```diff
@@ -1,465 +1,180 @@
+                .into_iter()
+                .map(Into::into)
+                .collect::<Vec<FetchedLogItem>>()
+        );
     });
 
-    // Ensure the logger server has time to start.
-    // TODO: find out why setting this lower causes spurious failures of these tests.
-    tokio::time::sleep(std::time::Duration::from_millis(500)).await;
-
-    // Start a subscriber and generate some logs using an instrumented deploy function.
-    generate_service_logs(port, DEPLOYMENT_ID.into(), deploy_instrumented);
-    tokio::time::sleep(std::time::Duration::from_millis(1000)).await;
-
-    let dst = format!("http://localhost:{port}");
-
-    let mut client = LoggerClient::connect(dst).await.unwrap();
-
-    // Get the generated logs
-    let response = client
-        .get_logs(Request::new(LogsRequest {
-            deployment_id: DEPLOYMENT_ID.into(),
-        }))
-        .await
-        .unwrap()
-        .into_inner();
-
-    let expected = vec![
-        MinLogItem {
-            level: LogLevel::Info,
-            fields: json!({"message": "[span] deploy_instrumented", "deployment_id": DEPLOYMENT_ID.to_string() }),
-        },
-        MinLogItem {
-            level: LogLevel::Error,
-            fields: json!({"message": "error"}),
-        },
-        MinLogItem {
-            level: LogLevel::Warn,
-            fields: json!({"message": "warn"}),
-        },
-        MinLogItem {
-            level: LogLevel::Info,
-            fields: json!({"message": "info", "deployment_id": DEPLOYMENT_ID.to_string()}),
-        },
-        MinLogItem {
-            level: LogLevel::Debug,
-            fields: json!({"message": "debug"}),
-        },
-        MinLogItem {
-            level: LogLevel::Trace,
-            fields: json!({"message": "trace"}),
-        },
-    ];
-
-    assert_eq!(
-        response
-            .log_items
-            .into_iter()
-            .map(MinLogItem::from)
-            .collect::<Vec<_>>(),
-        expected
-    );
-
-    // Generate some logs with a fn not instrumented with deployment_id.
-    generate_service_logs(port, DEPLOYMENT_ID.into(), deploy);
-
-    let response = client
-        .get_logs(Request::new(LogsRequest {
-            deployment_id: DEPLOYMENT_ID.into(),
-        }))
-        .await
-        .unwrap()
-        .into_inner();
-
-    // Check that no more logs have been recorded.
-    assert_eq!(
-        response
-            .log_items
-            .into_iter()
-            .map(MinLogItem::from)
-            .collect::<Vec<_>>(),
-        expected
-    );
+    tokio::select! {
+        _ = server => panic!("server stopped first"),
+        _ = test => ()
+    }
 }
 
 #[tokio::test]
-async fn generate_and_stream_logs() {
+async fn get_stream_logs() {
     let port = pick_unused_port().unwrap();
     let addr = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), port);
-    const DEPLOYMENT_ID: &str = "stream-logs-deployment-id";
+    const DEPLOYMENT_ID: &str = "runtime-fetch-logs-deployment-id";
```

Comment:
I would say let's make this a non-constant since having a constant in a scope does not make sense.

We can simply:

```rs
let deployment_id = "runtime-fetch-logs-deployment-id";
```